flood_adapt/object_model/tipping_point.py

```diff
@@ -0,0 +1,328 @@
+        operations = {"greater": lambda x, y: x >= y, "less": lambda x, y: x <= y}
+        return operations[operator](current_value, threshold)
+
+    # FUNCTIONS THAT ARE STILL NOT IMPLEMENTED - from benefits
+    def has_run_check(self):
+        """Check if the tipping point analysis has already been run"""
+        results_toml = self.results_path.joinpath("results.toml")
+        results_csv = self.results_path.joinpath("results.csv")
+        results_html = self.results_path.joinpath("results.html")
+        # TODO: check this
+
+        check = all(results_toml.exists() for result in [results_toml, results_csv])
+        if check:
+            with open(results_toml, "rb") as fp:
+                self.results = tomli.load(fp)
+
+    def check_scenarios(self):
+        """Check which scenarios are needed for this tipping point calculation
+        and if they have already been created"""
+        # calculating scenarios per slr level
+        scenarios_calc = {
+            f"slr_{slr}": {
+                "name": f"slr_{slr}",
+                "event": self.attrs.event_set,
+                "projection": self.attrs.projection,
+                "strategy": self.attrs.strategy,
+                "sealevelrise": slr,
+                "tipping_point_metric": self.attrs.tipping_point_metric,
+            }
+            for slr in self.attrs.sealevelrise
+        }
+
+        scenarios_avail = []
+        for scenario_path in list(
+            self.database_input_path.joinpath("scenarios").glob("*")
+        ):
+            scenarios_avail.append(
+                Scenario.load_file(scenario_path.joinpath(f"{scenario_path.name}.toml"))
+            )
+
+        # Check if any of the needed scenarios are already there
+        for scenario in scenarios_calc.keys():
+            scn_dict = scenarios_calc[scenario].copy()
+            scn_dict["name"] = scenario
+            scenario_obj = Scenario.load_dict(scn_dict, self.database_input_path)
+            created = [
+                scn_avl for scn_avl in scenarios_avail if scenario_obj == scn_avl
+            ]
+            if len(created) > 0:
+                scenarios_calc[scenario]["scenario created"] = created[0].attrs.name
+                if created[0].init_object_model().direct_impacts.has_run:
+                    scenarios_calc[scenario]["scenario run"] = True
+                else:
+                    scenarios_calc[scenario]["scenario run"] = False
+            else:
+                scenarios_calc[scenario]["scenario created"] = "No"
+                scenarios_calc[scenario]["scenario run"] = False
+
+        df = pd.DataFrame(scenarios_calc).T
+        self.scenarios = df.astype(
+            dtype={
+                "event": "str",
+                "projection": "str",
+                "strategy": "str",
+                "scenario created": "str",
+                "scenario run": bool,
+            }
+        )
+        return self.scenarios
```

Comment:
Im not convinced this is good behaviour. We are doing too much database related stuff in this class, which will produce headaches! Instead, this class should have standalone functions that the database can call after constructing the target_path (like load_file(target_path) and save(target_path) in other classes). This will decouple the database and tipping point and will make refactoring/testing/life in general much better as the tipping point now takes in any path!

A Tipping point object should be able to exist without a database object, which isnt possible now.
I think that creating a folder next to the tippingpoint.toml, that contains the input folders for all scenarios is nicer. The database can then just loop over whatever is in the tippingpoint input folder, and compare the tipping point scenarios to the ones stored in the database itself to see if the output can be copied, construct the result path, and save results) 

So, defer the running/checking/saving of the tippingpoint scenarios to the database, similar to Database.run_scenario(), which can then can call tippingpoint.evaluate(), save results etc.